app/controllers/admin/users_controller.rb

```diff
@@ -35,7 +36,9 @@ def update
     end
 
     def destroy
-      if @user.destroy
+      if @user == current_user
+        redirect_to admin_users_path, notice: t('.self_delete')
+      elsif @user.destroy
```

Comment:
使用 guard clause 表達你的意圖
```suggestion
      return redirect_to admin_users_path, notice: t('.self_delete') if @user == current_user

      if @user.destroy
```